pkg/proxy/client/client_conn.go

```diff
@@ -76,7 +77,7 @@ func (cc *ClientConnection) Run(ctx context.Context) {
 		return
 	}
 
-	if err := cc.processMsg(ctx); err != nil {
+	if err := cc.processMsg(ctx); err != nil && !errors.Is(err, io.EOF) {
```

Comment:
I need to know whether it's disconnected by the server or the client.
In many cases I encountered, the user reports that the client disconnects with the server and suspects it's the problem of the server. I need to check the log to confirm that it's the problem of the client because the client quits first.

Reply:
Got. I modified the code a little bit to wrap errors from `client` with `ErrClientConn`.

So if `EOF` comes from `client`, it is ignored. If it comes from `server`, it is logged.

Reply:
I think it's better to log even if `EOF` comes from the client. I can directly judge from one log file instead of searching all log files to confirm that there's no error from the server.

Reply:
>  I can directly judge from one log file instead of searching all log files to confirm that there's no error from the server.

That argument does not hold, if need to know if there is error from server. You need to check that all errors are from clients, in another words. You can just check if this connection is from server or client.

Whatever, I added `clientErr=boolean` and `serverErr=boolean` further. You can now `grep -r -H serverErr=true` to check that, without the need of logging `EOF` from the client.